### ISSUE TITLE
Visual Bug with portals crossing the world boundary

### ISSUE THREAD

**Issue by AutoCADEng** [qouteall/ImmersivePortalsModForForge#116]
Hi, I've discovered a bug in .20 forge.

Basically what I did was the following.

Dug a 3x3 hole until I world stacked into the nether.
Built a 3x3 vertical portal to move me into the overworld.
Doing this it's possible to travel to the far lands with 9 or less portals, depending on your starting point.The issue happens with the very last portal.
Portal no. 8 will be at 16 million. =8^8
Portal no.9 should be at 134 million(8^9), but since mine craft's world is limited to 30 million.
It's outside of bounds. When I got to the portal it would not render, and I could stand on top of it. When I broke the floor I traveled through it to 30 million, as expected.

However there was no portal generated in the over-world.

**Comment by LoganDark** [comment]
Portals can't render chunks outside the 30 million boundary because those chunks do not exist. And they don't load because they're too far outside the world border.

This is a (portal frame) generation bug, not a visual bug. Visually, things are fine. But a portal should not link up with something outside the border

**Comment by qouteall** [comment]
I never thought of that dimension stack can make the player easily reach the world border.

what do you mean "stand on top of it" and "broke the floor"?

**Comment by LoganDark** [comment]
> what do you mean "stand on top of it" and "broke the floor"?

Since the chunk wasn't loaded, there were no collisions (I assume). But they still collided with blocks below the portal for whatever reason. Then they finally teleported when their camera crossed the portal.

**Comment by AutoCADEng** [comment]
Yeah, basically.
I had to fall through the portal fully before I got teleported.

**Comment by LoganDark** [comment]
> I had to fall through the portal fully before I got teleported.

That's intended behavior. What's not intended is being able to stand on blocks behind the portal like this. Again, not a visual bug :P

**Comment by AutoCADEng** [comment]
Yeah I'm guessing it's because it hasn't properly connected to a portal on the other side. Hence it can't make heads of tails of whats on the other side, until the teleport is made and minecraft takes over.

**Comment by LoganDark** [comment]
> Yeah I'm guessing it's because it hasn't properly connected to a portal on the other side.

Portals are one-way and not linked in any way.

A bi-way portal is two portals, one on each side. And a regular nether portal has two portals per side, so you can enter the portal from both directions. For a total of 4 portals.

These portals don't care if there's another portal on the other side or not. It doesn't explain the collision problem at all.

If there was just no blocks on the other side, you would still be able to fall through the portal, so that doesn't explain it either.

Maybe it has something to do with being outside the border, or something to do with being in an unloaded chunk...

**Comment by IgeNiaI** [comment]
I suggest making it so that when you build a portal with stacked dimensions enabled, it will create one in the nether at the same coordinates as in overworld. Basically remove the 1 block in nether = 8 blocks in overworld.
